graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,55 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly.
+
+    Restrictions on use of directives:
+    - @output and @output_source are disallowed
+    - @macro_edge_target is not allowed to be inside a @fold scope
+
+    Args:
+        ast: GraphQL AST describing a subtree of the macro
```

Comment:
What's the exact object name?

Reply:
https://github.com/kensho-technologies/graphql-compiler/pull/210#discussion_r265748334